src/php/Command/Test/TestCommandOptions.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phel\Command\Test;
+
+final class TestCommandOptions
+{
+    private const FILTER_OPTION = '--filter=';
+
+    private string $filter = '';
+
+    public static function empty(): self
+    {
+        return new self();
+    }
+
+    public static function fromArray(array $options): self
+    {
+        $self = new self();
+
+        foreach ($options as $option) {
```

Comment:
I would recommend to use `getopt` (https://www.php.net/manual/de/function.getopt.php) or symfony/console  for parsing command line arguments. Everything else will probably fail (edge cases)